nncf/tensorflow/graph/converter.py

```diff
@@ -144,18 +344,66 @@ def _get_layer(self, layer_name: str) -> tf.keras.layers.Layer:
+            if weight_node_name is not None:
+                shared_node_dict = custom_layer_info.shared_weight_node_names_vs_weighted_op_node_names
+                is_shared = len(shared_node_dict[weight_node_name]) > 1
+            nncf_node = nncf_graph.add_nncf_node(node_name=node_info.target_node_name,
+                                                 node_type=node_info.node_type,
+                                                 node_metatype=node_info.node_metatype,
+                                                 # TODO (vshampor): collect layer attributes for custom nodes
+                                                 layer_attributes=None,
+                                                 layer_name=node_info.weight_node_name,  # sic!
+                                                 is_shared=is_shared,
+                                                 ignored_algorithms=['quantization'])
+            node_name_vs_nncf_node_ids[node_info.target_node_name] = nncf_node.node_id
+            self._nncf_node_names_vs_custom_layer_name[node_info.target_node_name] = custom_layer_name
+        for edge, edge_data in custom_layer_info.edge_infos.items():
+            from_node_name, to_node_name = edge
+            from_node_id = node_name_vs_nncf_node_ids[from_node_name]
+            to_node_id = node_name_vs_nncf_node_ids[to_node_name]
+            nncf_graph.add_edge_between_nncf_nodes(from_node_id, to_node_id,
+                                                   tensor_shape=edge_data.tensor_shape,
+                                                   input_port_id=edge_data.input_port_id,
+                                                   output_port_id=edge_data.output_port_id,
+                                                   dtype=edge_data.dtype)
+        return nncf_graph
+
+
+class TFModelConverterFactory:
+    @staticmethod
+    def create(model) -> TFModelConverter:
+        func_model = is_functional_model(model)
+        seq_model = is_sequential_model(model)
+
+        if not func_model and not seq_model:
+            RuntimeError('convert_keras_model_to_nncf_graph function supports '
+                         'only sequential or functional models')
+
+        if func_model:
+            converter = FunctionalConverter(model)
+        else:
+            converter = SequentialConverter(model)
+        return converter
+
+
 
 class FunctionalConverter(TFModelConverter):
     def __init__(self, model: tf.keras.Model):
         super().__init__(model)
-        self._raw_nodes = {}  # type: Dict[str, List[Dict]]
         self._layer_info = {}  # type: Dict[str, Dict]
         self._collect_layer_information()
-        self._node_info = {}  # type: Dict[str, Dict]
         self._layer_name_to_node_names = defaultdict(set)
         self._collect_node_information()
         self._edge_info = {}  # type: Dict[Tuple[str, str], Dict]
         self._collect_edge_information()
+        self._converted_graph = self._convert()
```

Comment:
I needed the converter object to hold the complete `NNCFGraph` so that I could build the target points using the same object. If moving the target point set creation to the `TFModelTransformer` works, I probably won't need to create the graph at `__init__`.

Reply:
Done